### DISCUSSION TITLE
Add ability to map OAuth users to Grafana organizations

### DISCUSSION THREAD

**Discussion by treydock** [grafana/grafana#39102] [General]
**What would you like to be added**:

The ability for OIDC authentication to map people to a organization based on the claims returned during authentication.  This would be very similar to `role_attribute_path`.

Example:

```
org_attribute_path = contains(groups[*], 'staff') && 1 ||  3
```
**Why is this needed**:

Right now the only way to do anything close to this is use `auto_assign_org_id` then for people who need a different org, assign them to that org manually.  This doesn't actually work, see #22605.

This feature request may also resolve #16763

**Comment by seanlaff** [comment]
Huge proponent of this. We rely heavily on OAuth across our apis and apps. In grafana's case we use it to control our users' login and grafana permissions via `role_attribute_path`. Additionally, we make heavy use of the `Forward OAuth Identity` feature in the datasource settings. 

![Screen Shot 2020-06-01 at 1 48 25 PM](https://user-images.githubusercontent.com/3194832/83438188-faa77f80-a40e-11ea-8c6a-d82b58b60934.png)

Overall grafana provides a fantastic experience allowing our identity provider to pass this info downstream to grafana without relying on flaky, intermittent api calls for synchronization... however the dream falls short once orgs enter the picture since grafana doesn't support reading them from the token.

Any way we can help with this change?

**Comment by aknuds1** [comment]
Thanks, looking into this.

**Comment by treydock** [comment]
This will be resolved once #23661 is merged.

**Comment by aknuds1** [comment]
@treydock Thanks, I realized in the meantime.

**Comment by thomas315** [comment]
> This will be resolved once #23661 is merged.

I think this issue won't get resolved as mentioned in related closed issue.

**Comment by kminehart** [comment]
Since it's been a year since the last comment on this issue, I'm going to move this to a discussion since there's so many reactions in the pull request / this issue. I'll add more information in the discussion.

**Comment by xenofree** [comment]
Is this planned for a future release ?

**Reply by geomtech** [reply]
We are interested too.

**Reply by umarkhan17** [reply]
We want this too. Does anyone has any  workarounds?

**Reply by benjamin-bergia** [reply]
Same here. I just finished setting up the oauth on my grafana instance and not being able to do anything else that sticking everybody in the same default org is a real bummer.

**Reply by tyspice** [reply]
I would also really love to see this feature added.

**Reply by despian** [reply]
+1 any news? We would also like the ability to grant OAuth users access to all organisations.

**Reply by Hinkepeet** [reply]
+1, would love to see this funtionality

**Reply by rafaribe** [reply]
We are also interested in this.

**Reply by philippebraxmeier** [reply]
We are interested too.

**Reply by kelseywa** [reply]
Any updates on this? Would like to see this functionality too!

**Reply by MioOgbeni** [reply]
Any updates? It is very usefull functionality. I still cannot believe that this is unsupported.

**Reply by Hinkepeet** [reply]
I doubt this will ever come to the OSS Version of Grafana.
It’s an enterprise feature called Team Sync, which allows this exact feature.
https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-team-sync/

**Reply by faxm0dem** [reply]
While very similar in scope, it's not exactly the same as the feature you mentioned is about Teams, not Organizations.
Nevertheless, I think you're right in saying that this probably makes it an Enterprise-only feature.

**Comment by beatkind** [comment]
@kminehart is there any update about this feature?

**Reply by ShakeelHussain** [reply]
Hi we just finished configuring OAuth and stucked here 
Any updates about this feature?

**Comment by faxm0dem** [comment]
Hi, we'd also very much like to see this feature added

**Comment by sveinf** [comment]
Great if this feature would be added

**Comment by hbjydev** [comment]
C'mon guys. I know you were trying to cut down on multi-org features but this is _basic_.

**Comment by mySCHOTT** [comment]
Hello dev team,
I would highly appreciate it, if [auth.azuread] supports multiple organizations directly defined within the Azure AD such as handling the roles viewer|editor|admin insite the app.registration. 
In m mind, it make sense to administrate users/groups, theier roles and other properties like grafana organization memberships complete within the azure ad.

**Comment by gelicia** [comment]
Hello! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

This discussion was converted into an issue, which is available [here](https://github.com/grafana/grafana/issues/73448).
